app/src/androidTest/java/com/example/bigowlapp/activity/SupervisedGroupListActivityTest.java

```diff
@@ -0,0 +1,129 @@
+                    "group00".concat(String.valueOf(i)).concat("fName"),
+                    "group00".concat(String.valueOf(i)).concat("lName"),
+                    "+12300".concat(String.valueOf(i)),
+                    "group00".concat(String.valueOf(i)).concat("@mail.com"),
+                    null
+            );
+            MutableLiveData<User> groupSupervisorData = new MutableLiveData<>(groupSupervisor);
+
+            List<String> groupSupervisedUserId = new ArrayList<>();
+            for (int j = 1; j < (int) ((Math.random() * 2) + 1); j++) {
+                groupSupervisedUserId.add(
+                        "group00".concat(String.valueOf(i)).concat("user00").concat(String.valueOf(j)));
+            }
+            groupSupervisedUserId.add("abc123");
+
+            Group newGroup = new Group(
+                    "group00".concat(String.valueOf(i)),
+                    "groupName00".concat(String.valueOf(i)),
+                    groupSupervisor.getUId(),
+                    groupSupervisedUserId
+            );
+            testUserSupervisedGroupList.add(newGroup);
+
+            when(userRepository.getDocumentByUId(groupSupervisor.getUId(), User.class))
+                    .thenReturn(groupSupervisorData);
+            when(supervisedGroupListViewModel.getSupervisor(groupSupervisor.getUId())).thenReturn(groupSupervisorData);
+        }
+        MutableLiveData<List<Group>> testUserSupervisedGroupListData = new MutableLiveData<>(testUserSupervisedGroupList);
+        testUserSupervisedGroupListData.postValue(testUserSupervisedGroupList);
+
+        when(testFirebaseUser.getUid()).thenReturn("abc123");
+        when(authRepository.getCurrentUser()).thenReturn(testFirebaseUser);
+        when(supervisedGroupListViewModel.getCurrentUserData()).thenReturn(testUserData);
+
+        when(groupRepository.getListOfDocumentByArrayContains(
+                anyString(), anyString(), eq(Group.class)))
+                .thenReturn(testUserSupervisedGroupListData);
+        when(supervisedGroupListViewModel.getSupervisedGroupListData()).thenReturn(testUserSupervisedGroupListData);
+
+        activityRule.getScenario().moveToState(Lifecycle.State.CREATED);
+        activityRule.getScenario().onActivity(activity -> {
+            activity.setSupervisedGroupListViewModel(supervisedGroupListViewModel);
+        });
+        activityRule.getScenario().moveToState(Lifecycle.State.RESUMED);
+    }
+
+    @Test
+    public void activityListOutUserSupervisedGroupsTest() {
+        verify(supervisedGroupListViewModel, atMostOnce()).getCurrentUserData();
+        verify(supervisedGroupListViewModel, atMostOnce()).getSupervisedGroupListData();
+
+        for (int i = 0; i < testUserSupervisedGroupList.size(); i++) {
+            assertEquals(testUserSupervisedGroupList.get(i),
+                    supervisedGroupListViewModel.getSupervisedGroupListData().getValue().get(i));
+        }
```

Comment:
You setup your mock to return the data you are checking. So this assert is kind of unnecessary or at most is testing that Mockito when statement works. Since our goal is to test the activity and not Mockito, I would say this can be removed.